### PR TITLE
Line-line intersection edge cases

### PR DESCRIPTION
**Pull request by simoncozens**
Literally edge cases, this deals with intersections on coincident lines.

### CODE REVIEW

**Comment by simoncozens** [comment]
There's still an issue with coincident lines which don't share a start or end point.

File: Tests/misc/bezierTools_test.py [hunk 1]
```diff
@@ -197,3 +198,13 @@ def test_intersections_linelike():
...
     assert pt == (0.0, 0.5)
+
+
+def test_intersections_samestart():
+    seg1 = [(250, 1000), (250, 810)]
+    seg2 = [(250, 1000), (250, 400)]
+    pt = lineLineIntersections(*seg1, *seg2)[0][0]
+    assert pt == (250.0, 1000.0)
+    seg3 = [(250, 810), (250, 500)]
+    assert lineLineIntersections(*seg1, *seg3) == []
```
[the remark this answers is not in the document]
Comment (anthrotype) [reply Tests/misc/bezierTools_test.py:209]: oh sorry I see now (was typing from mobile), the segments actually are coincident, sitting on top of one another. In theory they intersect in _all_ the points where they overlap, but since we need to return some finite results it's ok to return [] in this edge case

[the remark this answers is not in the document]
Comment (simoncozens) [reply Tests/misc/bezierTools_test.py:209]: Exactly.
